server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -381,6 +379,106 @@ func (s *BuildBuddyServer) JoinGroup(ctx context.Context, req *grpb.JoinGroupReq
 	return &grpb.JoinGroupResponse{}, nil
 }
 
+func (s *BuildBuddyServer) GetApiKeys(ctx context.Context, req *akpb.GetApiKeysRequest) (*akpb.GetApiKeysResponse, error) {
+	userDB := s.env.GetUserDB()
+	if userDB == nil {
+		return nil, status.UnimplementedError("Not Implemented")
+	}
+	groupID := req.GetGroupId()
+	if err := s.authorizeGroupAccess(ctx, groupID); err != nil {
+		return nil, err
+	}
+	tableKeys, err := userDB.GetAPIKeys(ctx, groupID)
+	if err != nil {
+		return nil, err
+	}
+	protoKeys := make([]*akpb.ApiKey, 0, len(tableKeys))
+	for _, k := range tableKeys {
+		protoKeys = append(protoKeys, &akpb.ApiKey{
+			Id:    k.APIKeyID,
+			Value: k.Value,
+			Label: k.Label,
+		})
+	}
+	return &akpb.GetApiKeysResponse{
+		ApiKey: protoKeys,
+	}, nil
+}
+
+func (s *BuildBuddyServer) CreateApiKey(ctx context.Context, req *akpb.CreateApiKeyRequest) (*akpb.CreateApiKeyResponse, error) {
+	userDB := s.env.GetUserDB()
+	if userDB == nil {
+		return nil, status.UnimplementedError("Not Implemented")
+	}
+	groupID := req.GetGroupId()
+	if err := s.authorizeGroupAccess(ctx, groupID); err != nil {
+		return nil, err
+	}
+	k, err := userDB.CreateAPIKey(ctx, groupID, req.GetLabel())
+	if err != nil {
+		return nil, err
+	}
+	return &akpb.CreateApiKeyResponse{
+		ApiKey: &akpb.ApiKey{
+			Id:    k.APIKeyID,
+			Value: k.Value,
+			Label: k.Label,
+		},
+	}, nil
+}
+
+func (s *BuildBuddyServer) authorizeAPIKeyWrite(ctx context.Context, apiKeyID string) error {
+	if apiKeyID == "" {
+		return status.InvalidArgumentError("API key ID is required")
+	}
+	user, err := perms.AuthenticatedUser(ctx, s.env)
+	if err != nil {
+		return err
+	}
+	userDB := s.env.GetUserDB()
+	if userDB == nil {
+		return status.UnimplementedError("Not Implemented")
+	}
+	// Check that the user belongs to the group that owns the requested API key.
+	key, err := userDB.GetAPIKey(ctx, apiKeyID)
+	if err != nil {
+		return err
+	}
+	acl := perms.ToACLProto( /* userID= */ nil, key.GroupGroupID, key.Perms)
+	return perms.AuthorizeWrite(user, acl)
+}
+
+func (s *BuildBuddyServer) UpdateApiKey(ctx context.Context, req *akpb.UpdateApiKeyRequest) (*akpb.UpdateApiKeyResponse, error) {
+	userDB := s.env.GetUserDB()
+	if userDB == nil {
+		return nil, status.UnimplementedError("Not Implemented")
+	}
+	if err := s.authorizeAPIKeyWrite(ctx, req.GetId()); err != nil {
+		return nil, err
+	}
+	if err := userDB.UpdateAPIKey(ctx, &tables.APIKey{
+		APIKeyID: req.GetId(),
+		Label:    req.GetLabel(),
+	}); err != nil {
+		return nil, err
+	}
```

Comment:
super uber nit: for readability's sake I think it's nice to keep the if cases to one line as much as possible. In this case you could do:

tk := &tables.APIKey{
	APIKeyID: req.GetId(),
	Label:    req.GetLabel(),
}
if err := userDB.UpdateAPIKey(ctx, tk); err != nil {
  return nil, err
}

Reply:
Much cleaner - done